internal/fs/inode/dir.go

```diff
@@ -771,15 +771,19 @@ func (d *dirInode) CreateChildFile(ctx context.Context, name string) (*Core, err
 	}, nil
 }
 
-func (d *dirInode) CreateLocalChildFile(name string) (*Core, error) {
+func (d *dirInode) CreateLocalChildFile(name string, updateInodeFn func(core Core) error) error {
```

Comment:
Did you consider the scenarios of updating it here vs updating the type-cache when the sync to GCS is done.
for eg: create file (not synced to GCS), delete, create dir with same name?

Reply:
We don't get parent Inode in sync/flush op.

Reply:
As discussed with @vadlakondaswetha , we can update the type-cache when the temp-file gets created. If the file gets deleted, DeleteFile will be called where we remove it from the cache.